plugins/parquet-data-format/src/main/java/com/format/parquet/ParquetDataFormatPlugin.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package com.format.parquet;
+
+import org.opensearch.execution.search.spi.DataFormatCodec;
+import org.opensearch.index.engine.exec.format.DataFormat;
+import org.opensearch.plugins.DataFormatPlugin;
+import org.opensearch.plugins.Plugin;
+
+import java.util.Map;
+import java.util.Optional;
+
+import com.format.parquet.engine.ParquetDataFormat;
+
+/**
+ * OpenSearch plugin that provides Parquet data format support for indexing and query operations.
+ */
+public class ParquetDataFormatPlugin extends Plugin implements DataFormatPlugin {
```

Comment:
i'm still struggling with the need for this plugin at a top level - can we not have dataformat pieces as a lib inside of engine specific plugins, ie engines provide indexer & searcher functionality directly.